src/network/netdev/bond.c

```diff
@@ -371,6 +390,115 @@ int config_parse_arp_ip_target_address(const char *unit,
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse actor system priority '%s', ignoring: %m", rvalue);
+                return 0;
+        }
+
+        if (v == 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse actor system priority '%s'. Range is [1,65535], ignoring: %m", rvalue);
+                return 0;
+        }
+
+        b->ad_actor_sys_prio = v;
+
+        return 0;
+}
+
+int config_parse_ad_user_port_key(const char *unit,
+                                  const char *filename,
+                                  unsigned line,
+                                  const char *section,
+                                  unsigned section_line,
+                                  const char *lvalue,
+                                  int ltype,
+                                  const char *rvalue,
+                                  void *data,
+                                  void *userdata) {
+        Bond *b = userdata;
+        uint16_t v;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = safe_atou16(rvalue, &v);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse user port key '%s', ignoring: %m", rvalue);
+                return 0;
+        }
+
+        if (v > 1023) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse user port key '%s'. Range is [0,1023], ignoring: %m", rvalue);
+                return 0;
+        }
+
+        b->ad_user_port_key = v;
+
+        return 0;
+}
+
+int config_parse_ad_actor_system(const char *unit,
+                                 const char *filename,
+                                 unsigned line,
+                                 const char *section,
+                                 unsigned section_line,
+                                 const char *lvalue,
+                                 int ltype,
+                                 const char *rvalue,
+                                 void *data,
+                                 void *userdata) {
+        Bond *b = userdata;
+        _cleanup_free_ struct ether_addr *n = NULL;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        n = new0(struct ether_addr, 1);
+        if (!n)
+                return log_oom();
+
+        r = ether_addr_from_string(rvalue, n);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Not a valid MAC address %s. Ignoring assignment: %m", rvalue);
+                return 0;
+        }
+
+        if (ether_addr_is_null(n) || (n->ether_addr_octet[0] & 0x01)) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Not a valid MAC address %s, can not be null or multicast. Ignoring assignment: %m", rvalue);
+                return 0;
+        }
+
+        b->ad_actor_system = TAKE_PTR(n);
```

Comment:
It seems that the previous value should be freed before the assignment. Otherwise, it causes a memory leak when `AdActorSystem` is specified at least two times:
```sh
=================================================================
==7215==ERROR: LeakSanitizer: detected memory leaks
Direct leak of 6 byte(s) in 1 object(s) allocated from:
    #0 0x7f0107e88e50 in calloc (/lib64/libasan.so.5+0xeee50)
    #1 0x557595c93acd in config_parse_ad_actor_system ../src/network/netdev/bond.c:482
    #2 0x7f0106cafacc  (/home/vagrant/systemd/build/src/shared/libsystemd-shared-239.so+0x720acc)
    #3 0x7f0106cb0e77  (/home/vagrant/systemd/build/src/shared/libsystemd-shared-239.so+0x721e77)
    #4 0x7f0106cb1d89  (/home/vagrant/systemd/build/src/shared/libsystemd-shared-239.so+0x722d89)
    #5 0x557595bdb16c in netdev_load_one ../src/network/netdev/netdev.c:662
    #6 0x557595bdd098 in netdev_load ../src/network/netdev/netdev.c:765
    #7 0x557595b92d01 in manager_load_config ../src/network/networkd-manager.c:1495
    #8 0x557595b7d924 in main ../src/network/networkd.c:90
    #9 0x7f01052b311a in __libc_start_main (/lib64/libc.so.6+0x2311a)
SUMMARY: AddressSanitizer: 6 byte(s) leaked in 1 allocation(s).
```

Reply:
I'd also add something like
```diff
diff --git a/test/test-network/conf/25-bond.netdev b/test/test-network/conf/25-bond.netdev
index 3ed79282a..1615b443e 100644
--- a/test/test-network/conf/25-bond.netdev
+++ b/test/test-network/conf/25-bond.netdev
@@ -14,3 +14,4 @@ MinLinks=1
 AdActorSysPrio=1218
 AdUserPortKey=811
 AdActorSystem=00:11:22:33:44:55
+AdActorSystem=00:11:22:33:44:55
```
so that it would be a little bit easier to catch it in the future.

Reply:
Ah, indeed. Please use `free_and_replace()` here.

Reply:
By the way, @yuwata do you know if anyone has ever started to work on a fuzzer that would be similar to `fuzz-unit-file` but for `systemd-networkd`? I'm pretty sure it'd be useful and certainly would be much more capable of generating nonsensical configs than me :-)

Reply:
@evverx I guess no one is working on that.

Reply:
@yuwata that's a bummer. Anyway, in the meantime, until the fuzzer is written, just out of curiosity, I ran a perl script that duplicate each line in the configs to see how `networkd` reacts to it. I'll open an issue with my findings attached shortly :-)

Reply:
I've just opened https://github.com/systemd/systemd/issues/10615.